From b940bc81a1c49db02fbde1cafdd9cdc162ef1cbc Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:07:09 -0400
Subject: [tools] perf: Move u64_swap union

Message-id: <1410181752-25631-185-git-send-email-jolsa@redhat.com>
Patchwork-id: 91869
O-Subject: [PATCH RHEL7.1 BZ1134356 184/307] perf tools: Move u64_swap union
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 88080ce7f6af1ad99ad4b2825938411975910116
Author: Borislav Petkov <bp@suse.de>
Date: Mon Apr 14 17:38:39 2014 +0200

description
===========
... to its single user's header, evsel.h.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/evsel.h b/tools/perf/util/evsel.h
index 0c9926c..0c581d0 100644
--- a/tools/perf/util/evsel.h
+++ b/tools/perf/util/evsel.h
@@ -91,6 +91,11 @@ struct perf_evsel {
  char   *group_name;
 };
 
+union u64_swap {
+ u64 val64;
+ u32 val32[2];
+};
+
 #define hists_to_evsel(h) container_of(h, struct perf_evsel, hists)
 
 struct cpu_map;
diff --git a/tools/perf/util/types.h b/tools/perf/util/types.h
index c51fa6b..5f3689a 100644
--- a/tools/perf/util/types.h
+++ b/tools/perf/util/types.h
@@ -16,9 +16,4 @@ typedef signed short    s16;
 typedef unsigned char    u8;
 typedef signed char    s8;
 
-union u64_swap {
- u64 val64;
- u32 val32[2];
-};
-
 #endif /* __PERF_TYPES_H */
-- 
1.7.1